src/core/providers/ogr/qgsogrprovidermetadata.cpp

```diff
@@ -1071,6 +1075,118 @@ bool QgsOgrProviderMetadata::saveLayerMetadata( const QString &uri, const QgsLay
   throw QgsNotSupportedException( QObject::tr( "Storing metadata for the specified uri is not supported" ) );
 }
 
+QList<QgsLayerMetadataProviderResult> QgsOgrProviderMetadata::searchLayerMetadata( const QString &uri, const QString &searchString, const QgsRectangle &geographicExtent, QgsFeedback *feedback )
+{
+
+  QList<QgsLayerMetadataProviderResult> results;
+  QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "ogr" ) ) };
```

Comment:
I don't understand the intention here... can't we just directly use "this" instead of retrieving the same metadata object from the registry?

Reply:
you are 100% right, I probably moved the logic from another class while refactoring.